pkg/apis/certmanager/v1alpha1/types_issuer.go

```diff
@@ -204,6 +207,29 @@ type ACMEIssuerDNS01ProviderAcmeDNS struct {
 	AccountSecret SecretKeySelector `json:"accountSecretRef"`
 }
 
+// ACMEIssuerDNS01ProviderRFC2136 is a structure containing the
+// configuration for RFC2136 DNS
+type ACMEIssuerDNS01ProviderRFC2136 struct {
+	// The IP address of the DNS supporting RFC2136. Required.
+	Nameserver string `json:"nameserver"`
+
+	// The name of the secret containing the TSIG value.
+	// If ``tsigSecretSecretRef`` is not defined, ``tsigKey`` is ignored.
+	// +optional
+	TSIGSecret SecretKeySelector `json:"tsigSecretSecretRef"`
+
+	// The TSIG Key name configured in the DNS. If ``tsigKey`` is not defined,
+	// ``tsigSecretSecretRef`` is ignored
+	// +optional
+	TSIGKey string `json:"tsigKey"`
```

Comment:
None of the other DNS providers currently provide this option (allowing users to specify the key directly on the resource).

I think for the time being it's best to remove this field - there's some ongoing discussion on how we should handle credentials more generally, but I think we shouldn't make this provider different to others 😄

Reply:
@munnerz oh, that's not the secret itself but the tsigkey *name*.  Maybe we can change the name to reflect that. The documentation reads: 
>  The TSIG Key name configured in the DNS